File: 1652-minimum-suffix-flips/1652-minimum-suffix-flips.rb
# @param {String} target
# @return {Integer}
def min_flips(target)
    current_state = "0"
    count = 0
    
    target.each_char do |char|
      if char != current_state
        count += 1
        current_state = char
      end
    end
    count
end

File: 1652-minimum-suffix-flips/1652-minimum-suffix-flips.rs
impl Solution {
  pub fn min_flips(target: String) -> i32 {
      let mut current_state = '0';
      let mut count = 0;

      for char in target.chars() {
          if char != current_state {
              count += 1;
              current_state = char;
          }
      }
      count
  }
}